modules/ballerina-http/src/main/java/org/ballerinalang/net/http/nativeimpl/response/Respond.java

```diff
@@ -76,8 +82,17 @@
             // default behaviour: keepAlive = true
             responseMessage.setHeader(Constants.CONNECTION_HEADER, Constants.HEADER_VAL_CONNECTION_KEEP_ALIVE);
         }
-        if (outboundResponseStruct.getRefField(Constants.RESPONSE_HEADERS_INDEX) != null) {
-            HttpUtil.setHeadersToTransportMessage(responseMessage, outboundResponseStruct);
+        BStruct entity = (BStruct) outboundResponseStruct.getNativeData(MESSAGE_ENTITY);
+        if (entity == null) {
+            entity = ConnectorUtils.createAndGetStruct(context,
+                    org.ballerinalang.mime.util.Constants.PROTOCOL_PACKAGE_MIME,
+                    org.ballerinalang.mime.util.Constants.ENTITY);
+            entity.setRefField(ENTITY_HEADERS_INDEX, new BMap<>());
+            outboundResponseStruct.addNativeData(MESSAGE_ENTITY, entity);
+            outboundResponseStruct.addNativeData(IS_ENTITY_BODY_PRESENT, false);
+        }
+        if (entity.getRefField(ENTITY_HEADERS_INDEX) != null) {
+            HttpUtil.setHeadersToTransportMessage(responseMessage, outboundResponseStruct, entity);
```

Comment:
Can we remove entity parameter as we can take it from outboundResponseStruct using outboundResponseStruct.getNativeData(MESSAGE_ENTITY) ?